src/L.Control.Locate.js

```diff
@@ -361,49 +365,51 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
         },
 
         /**
-         * This method is called when the user clicks on the control.
+         * This method is called when the user takes action on the control.
          */
-        _onClick: function() {
-            this._justClicked = true;
-            var wasFollowing =  this._isFollowing();
-            this._userPanned = false;
-            this._userZoomed = false;
-
-            if (this._active && !this._event) {
-                // click while requesting
-                this.stop();
-            } else if (this._active && this._event !== undefined) {
-                var behaviors = this.options.clickBehavior;
-                var behavior = behaviors.outOfView;
-                if (this._map.getBounds().contains(this._event.latlng)) {
-                    behavior = wasFollowing ? behaviors.inView : behaviors.inViewNotFollowing;
-                }
-
-                // Allow inheriting from another behavior
-                if (behaviors[behavior]) {
-                    behavior = behaviors[behavior];
-                }
-
-                switch (behavior) {
-                    case 'setView':
-                        this.setView();
-                        break;
-                    case 'stop':
-                        this.stop();
-                        if (this.options.returnToPrevBounds) {
-                            var f = this.options.flyTo ? this._map.flyToBounds : this._map.fitBounds;
-                            f.bind(this._map)(this._prevBounds);
-                        }
-                        break;
-                }
-            } else {
-                if (this.options.returnToPrevBounds) {
-                  this._prevBounds = this._map.getBounds();
+        _onButtonAction: function(e) {
+            if (e.type === 'click' || e.key === "Enter" ){
```

Comment:
I have not seen this pattern before. Can you point to some code in Leaflet that does the same for the standard buttons?

Reply:
Removed